### PR TITLE
PWN-3072 - Swap analytics: Swap_User_Confirmed Swap_Started Swap_Completed

### PR DESCRIPTION
**Pull request by gslevinkov**
https://p2pvalidator.atlassian.net/browse/PWN-3072

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/common/analytics/TrackerContract.kt [hunk 1]
```diff
@@ -1,18 +1,14 @@
...
-    fun regenerateDeviceId() = Unit
+interface TrackerContract {
+    fun logEvent(event: String, params: Map<String, Any> = emptyMap()) = NoOp
+    fun logEvent(event: String, params: Array<out Pair<String, Any>>) = NoOp
+    fun setUserProperty(key: String, value: String) = NoOp
+    fun incrementUserProperty(property: String, byValue: Int = 1) = NoOp
+    fun setUserPropertyOnce(key: String, value: String) = NoOp
+    fun setUserId(userId: String?) = NoOp
+    fun appendToArray(property: String, value: Int) = NoOp
+    fun regenerateDeviceId() = NoOp
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/common/analytics/TrackerContract.kt:13]: These looks very weird. 
Let's leave as `fun someName()` without any type? 
I thought we are added `NoOp` for functions realization

File: app/src/main/java/org/p2p/wallet/swap/ui/orca/OrcaSwapPresenter.kt [hunk 2]
```diff
@@ -282,6 +282,15 @@ class OrcaSwapPresenter(
                 destinationAmountUsd = destinationAmountUsd.toString()
             )
             swapAnalytics.logSwapVerificationInvoked(AuthAnalytics.AuthType.BIOMETRIC)
+            swapAnalytics.logSwapUserConfirmed(
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/swap/ui/orca/OrcaSwapPresenter.kt:285]: We already have event for this above. 
Maybe we should remove one of them?

File: app/src/main/java/org/p2p/wallet/common/analytics/TrackerContract.kt (hunk 1 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/common/analytics/TrackerContract.kt:13]: no need to ask me, it was just `= Unit` before me. i'll remove NoOp

File: app/src/main/java/org/p2p/wallet/swap/analytics/SwapAnalytics.kt [hunk 3]
```diff
@@ -211,6 +214,75 @@ class SwapAnalytics(private val trackerContract: TrackerContract) {
...
+        trackerContract.logEvent(
+            event = SWAP_USER_CONFIRMED,
+            params = mapOf(
+                "Token_A_Name" to tokenAName,
+                "Token_B_Name" to tokenBName,
+                "Swap_Sum" to swapSum,
+                "Swap_MAX" to if (isSwapMax) "True" else "False",
+                "Swap_USD" to swapUsd.toString(),
+                "Price_Slippage" to priceSlippage.toString(),
+                "Fees_Source" to feesSource.title
```
[the remark this answers is not in the document]
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/swap/analytics/SwapAnalytics.kt:235]: im not sure that this is a good idea. 
i think these event keys can be changed separately some day -  for example:
SWAP_USER_CONFIRMED will have `Swap_Sum_confirmed`
but SWAP_STARTED will have `Swap_Sum`
i think all these strings in static vars will be kept in memory while the app is running

File: app/src/main/java/org/p2p/wallet/swap/ui/orca/OrcaSwapPresenter.kt (hunk 2 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/swap/ui/orca/OrcaSwapPresenter.kt:285]: i'll ask the analyst

File: app/src/main/java/org/p2p/wallet/common/analytics/AmplitudeTracker.kt [hunk 4]
```diff
@@ -26,16 +27,20 @@ class AmplitudeTracker(app: Application) : TrackerContract {
...
-            try {
-                amplitude.logEvent(event, JSONObject(params.toMap()))
-            } catch (e: NullPointerException) {
-                Timber.w(e, "Unable to put key - value into json")
-            }
+            return
         }
+        try {
+            amplitude.logEvent(event, JSONObject(params))
+        } catch (e: NullPointerException) {
```
[the remark this answers is not in the document]
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/common/analytics/AmplitudeTracker.kt:37]: i can't see any JSONException in the internals

File: app/src/main/java/org/p2p/wallet/auth/analytics/OnBoardingAnalytics.kt [hunk 5]
```diff
@@ -1,34 +1,32 @@
...
-    private val tracker: TrackerContract
-) {
+import org.p2p.wallet.common.analytics.Analytics
+import org.p2p.wallet.common.analytics.constants.EventNames.ONBOARD_BACKING_UP_COPYING
+import org.p2p.wallet.common.analytics.constants.EventNames.ONBOARD_BACKING_UP_RENEW
+import org.p2p.wallet.common.analytics.constants.EventNames.ONBOARD_BACKING_UP_SAVING
+import org.p2p.wallet.common.analytics.constants.EventNames.ONBOARD_BACKUP_ERROR
+import org.p2p.wallet.common.analytics.constants.EventNames.ONBOARD_BACKUP_MANUALLY
+import org.p2p.wallet.common.analytics.constants.EventNames.ONBOARD_BIO_APPROVED
+import org.p2p.wallet.common.analytics.constants.EventNames.ONBOARD_BIO_REJECTED
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/analytics/OnBoardingAnalytics.kt:10]: maybe we should allow wildcard imports for analytics?

File: app/src/main/java/org/p2p/wallet/App.kt [hunk 6]
```diff
@@ -15,8 +15,7 @@ import org.p2p.wallet.auth.AuthModule
 import org.p2p.wallet.common.analytics.AnalyticsModule
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/App.kt:15]: tl;tr
- moved classes to corresponding packages
- removed unnecessary inheritance
- renamed classes to match their responsibilty

File: app/src/main/java/org/p2p/wallet/swap/analytics/SwapAnalytics.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/swap/analytics/SwapAnalytics.kt:235]: we've decided to leave it this way

**Review by 1sakov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/auth/analytics/OnBoardingAnalytics.kt (hunk 5 above)
Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/auth/analytics/OnBoardingAnalytics.kt:10]: But we have autoformat in our config.xml, are you experiencing some problems with that?

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/auth/analytics/OnBoardingAnalytics.kt:10]: no, im just talking about that with analytics we can't experience namespace conflicts, but when using wildcard import we can reduce import code a lot!

**Pull request merged**